### PR TITLE
Sarahmood/issue53

### LINKED ISSUE #53: GET /api/user && GET /api/organization/:id
For more information about these endpoints, please visit this [notion page](https://sehebi.notion.site/Define-Endpoints-7d0644dbe9174a41b3b44cf82476e60d).

### PR DESCRIPTION
**Pull request by sarahmood**
## The following were added

GET user / organization routes.
**Note :**  **I separated profiles to be public and private profiles.**

validation
swagger docs

### CODE REVIEW

File: src/docs/organizations.yaml [hunk 1]
```diff
@@ -0,0 +1,691 @@
...
+paths:
+    /api/organization/:id:
+        get:
+            tags:
+                - Organization
+            summary: View organization public profile
+            description: show public information of an organization
+            operationId: getOrganization
+            consumes:
+                - multipart/form-data
```
Comment (muhammed-shihebi) [inline src/docs/organizations.yaml]: Why multipart/form-data? 
Your endpoint consumes application/json actually.

File: src/docs/organizations.yaml [hunk 2]
```diff
@@ -0,0 +1,691 @@
...
+                            - Life Below Water
+                            - Life On Land
+                            - Peace
+                            - Youth
+                            - Justice
+                    example: [No Poverty, Zero Hunger]
+                followers:
+                    type: array
+                    items:
+                        $ref: '#/components/schemas/ShortUser'
```
Comment (muhammed-shihebi) [inline src/docs/organizations.yaml:221]: You are using the short user here, but it includes user email and in your endpoint, you are returning only 'id', 'firstName', 'lastName', and 'profileImage'. Consider defining a new schema for your short user.

File: src/docs/organizations.yaml [hunk 3]
```diff
@@ -0,0 +1,691 @@
...
+            produces:
+                - application/json
+            parameters:
+                - name: id
+                  in: path
+                  required: true
+                  schema:
+                      type: string
+                      example: 5e9f8f8f8f8f8f8f8f8f8f8
+            responses:
```
Comment (muhammed-shihebi) [inline src/docs/organizations.yaml:23]: Could you please add the response code of the validation error.

File: src/docs/organizations.yaml [hunk 4]
```diff
@@ -0,0 +1,691 @@
...
+                    content:
+                        application/json:
+                            schema:
+                                $ref: '#/components/schemas/FullPublicOrganization'
+                404:
+                    description: Organization not found  or has an invalid value
+                    content:
+                        application/json:
+                            schema:
+                                $ref: '#/components/schemas/ValidationError'
```
Comment (muhammed-shihebi) [inline src/docs/organizations.yaml]: You can't use $ref and properties at the same time. If you look at your endpoint in swagger, you will see that only the $ref is visible, and non of the stuff in the properties is visible. 

Also, you are referencing the ValidationError, which has a different schema than yours. You have `{message: 'Page not found'}`.

So, you should add both your 404 error and the 422, which is the validation error.

File: src/docs/organizations.yaml [hunk 5]
```diff
@@ -0,0 +1,691 @@
...
+                    description: Internal server error
+    /api/organization/:
+        get:
+            tags:
+                - Organization
+            summary: View organization private profile
+            description: show private information of an organization
+            operationId: getOrganization
+            consumes:
+                - multipart/form-data
```
Comment (muhammed-shihebi) [inline src/docs/organizations.yaml]: please use application/json instead

File: src/docs/users.yaml [hunk 6]
```diff
@@ -0,0 +1,1128 @@
...
+                    description: Internal server error
+    /api/user/:
+        get:
+            tags:
+                - User
+            summary: View user private profile
+            description: show private information of a user
+            operationId: getUser
+            consumes:
+                - multipart/form-data
```
Comment (muhammed-shihebi) [inline src/docs/users.yaml]: application/json

File: src/docs/users.yaml [hunk 7]
```diff
@@ -0,0 +1,1128 @@
...
+paths:
+    /api/user/:id:
+        get:
+            tags:
+                - User
+            summary: View user public profile
+            description: show public information of a user
+            operationId: getUser
+            consumes:
+                - multipart/form-data
```
Comment (muhammed-shihebi) [inline src/docs/users.yaml]: application/json

**Review by muhammed-shihebi — CHANGES_REQUESTED** [review]
Great work @sarahmood as always, just few tweaks and it is ready to be merged.

File: src/docs/organizations.yaml (hunk 2 above)
Comment (sarahmood) [reply src/docs/organizations.yaml:221]: @muhammed-shihebi based on what we discussed, we said to take off email form all users in the project. I fixed that in dev branch and when I merge there will be no email in ShortUser.

Comment (sarahmood) [reply src/docs/organizations.yaml:221]: okay it is not removed haha! i will fix ShortUser in this PR to not have email

**Comment by halitbatur** [comment]
@sarahmood any updates on the changes requested, when do you think you will be able to fix it

**Review by muhammed-shihebi — APPROVED** [review]

**Review by halitbatur — APPROVED** [review]

**Pull request merged**